server/feeders/sensorsgenerator.js

```diff
@@ -55,27 +55,22 @@ var feedIDASSensors = function() {
 
   // Limit the number of calls to be done in parallel
   var q = async.queue(function(task, callback) {
-    idas.registerSensor(task.name, task.type)
-      .then(
-        function(response) {
-          sensorsAdded++;
-          return idas.initializeSensor(task.name, task.type)
-            .delay(1000) // This is to avoid clogging iotagent with requests.
-            .then(
-              function(response) {
-                sensorsInitialized++;
-              },
-              function(error) {
-                console.log('initializeSensor Error:', error);
-              });
-        },
-        function(error) {
-          console.log('createSensor Error:', error);
-        })
-      .done(
-        function(response) {
-          callback();
-        });
+    idas.registerSensor(task.restaurant, task.room, task.type)
+      .then(function(response) {
+        sensorsAdded++;
+        return idas.initializeSensor(task.restaurant, task.room, task.type)
+          .delay(1000) // This is to avoid clogging iotagent with requests.
+          .then(function(response) {
+            sensorsInitialized++;
+          }, function(error) {
+            console.log('initializeSensor Error:', error);
+          });
+      }, function(error) {
+        console.log('createSensor Error:', error);
+      })
+      .done(function(response) {
```

Comment:
I think the code above can simplified a lot

a.then(function() {
  return b();
}).then(function() {
  return c();
}).then(function() {
   finish();
}).catch(function(err) {
})